storm-core/src/ui/public/js/visualization.js

```diff
@@ -16,434 +16,330 @@
-        for(var i = 0; i < v[":inputs"].length; i++) {
-            if(stream_checked(v[":inputs"][i][":sani-stream"]) 
-               && v[":inputs"][i][":component"] == component)
-                ret = true;
+
+        function parseEdge(edgeJson, sourceId) {
+            // Make this stream available
+            addAvailableStream(edgeJson[":stream"])
+
+            // Create a unique Id
+            var id = edgeJson[":component"] + ":" + sourceId + ":" + edgeJson[":stream"];
+
+            // Determine if stream is enabled
+            if (!isStreamEnabled(edgeJson[":stream"])) {
+                // Remove edge
+                edges.remove({ "id": id });
+                return
+            }
+
+            if (!edges.get(id)) {
+                console.log("Updating edge " + id );
+                edges.update({
+                    "id": id,
+                    "from": edgeJson[":component"],
+                    "to": sourceId,
+                    "label": edgeJson[":stream"],
+                    "title": "From: " + edgeJson[":component"] + "<br>To: " + sourceId + "<br>Grouping: " + edgeJson[":grouping"]
+                });
+            }
         }
-    });
-    return ret;
-}
-
-function gather_stream_count(stats, stream, time) {
-    var transferred = 0;
-    if(stats)
-        for(var i = 0; i < stats.length; i++) {
-            if(stats[i][":transferred"] != null && stats[i][":transferred"][time] != undefined)
-            {
-                var stream_trans = stats[i][":transferred"][time][stream];
-                if(stream_trans != null)
-                    transferred += stream_trans;
+
+        function addAvailableStream(streamId) {
+            // Create a master list of all streams
+            if (availableStreamsHash[streamId] == null) {
+                availableStreamsHash[streamId] = true;
+                updateAvailableStreams();
             }
         }
-    return transferred;
-}
-
-
-function rechoose(jdat, sys, box) {
-    var id = box.id;
-    if($(box).is(':checked'))
-    {
-        //Check each node in our json data to see if it has inputs from or outputs to selected streams. If it does, add a node for it.
-        $.each(jdat,function(k,v) {
-            if( has_checked_stream_input(v[":inputs"]) || has_checked_stream_output(jdat, k))
-                sys.addNode(k,v);
-        });
-           
-        //Check each node in our json data and add necessary edges based on selected components.
-        $.each(jdat, function(k, v) {
-            for(var i = 0; i < v[":inputs"].length; i++)
-                if(v[":inputs"][i][":sani-stream"] === id) {
-                    
-                    sys.addEdge(v[":inputs"][i][":component"], k, v);
+
+        function updateAvailableStreams() {
+            var container = jQuery("#available-streams");
+            for (var streamName in availableStreamsHash) {
+                var entry = jQuery(container).find("#stream-" + streamName)
```

Comment:
Ah I've never played with Trident, I'll dig into that and see what is going on and sanitize them where needed.  Thanks for the heads up. 👍 

I'll also dig into making the sizing of the slide outs dynamic.